app/entry.server.tsx

```diff
@@ -1,34 +1,122 @@
-import { CacheProvider } from "@emotion/react"
-import { createStylesServer, injectStyles } from "@mantine/remix"
-import type { EntryContext } from "@remix-run/node"
+/**
+ * By default, Remix will handle generating the HTTP Response for you.
+ * You are free to delete this file if you'd like to, but if you ever want it revealed again, you can run `npx remix reveal` ✨
+ * For more information, see https://remix.run/file-conventions/entry.server
+ */
+
+import { PassThrough } from "node:stream"
+
+import type { AppLoadContext, EntryContext } from "@remix-run/node"
+import { createReadableStreamFromReadable } from "@remix-run/node"
 import { RemixServer } from "@remix-run/react"
-import { renderToString } from "react-dom/server"
+import { isbot } from "isbot"
+import { renderToPipeableStream } from "react-dom/server"
 import { getClientEnv } from "~/utils/environment.server"
-import { emotionCache } from "~/utils/mantineCache"
 
 global.ENV = getClientEnv()
 
-const server = createStylesServer(emotionCache)
+const ABORT_DELAY = 5_000
 
 export default function handleRequest(
   request: Request,
   responseStatusCode: number,
   responseHeaders: Headers,
   remixContext: EntryContext,
+  loadContext: AppLoadContext,
 ) {
-  // Wrap the server in the cache provider and set the cache
-  let markup = renderToString(
-    <CacheProvider value={emotionCache}>
-      <RemixServer context={remixContext} url={request.url} />
-    </CacheProvider>,
-  )
-  if (process.env.NODE_ENV !== "production") {
-    responseHeaders.set("Cache-Control", "no-store")
-  }
-  responseHeaders.set("Content-Type", "text/html")
-
-  return new Response(`<!DOCTYPE html>${injectStyles(markup, server)}`, {
-    status: responseStatusCode,
-    headers: responseHeaders,
+  return isbot(request.headers.get("user-agent"))
```

Comment:
I feel like im missing something here. There doesnt seem to be a difference between handleBotRequest and handleBrowserRequest. 

we dont need to complicate this PR more, but I think we should discuss this more. I have some thoughts and questions like:

- if they are the same why duplicate the code.
- if we duplicate the code maybe we should actually leverage it better.
- like maybe would we be able to block bots from all user pages, but allow them on api routes.
- maybe this logic is another place we could use to stop new users spamming us.

Reply:
I generated a new Remix template and took this file from there.. also see here https://github.com/mantinedev/remix-template/blob/master/app/entry.server.tsx

This can be removed actually (see comment on top) and I only had to bring it in to add this
https://github.com/pokt-foundation/portal-platform/blob/0b1f17b160dc67faf72f1a420c163b2087c271c6/app/entry.server.tsx#L16

Reply:
But yeah, your arguments are valid for the bot part and we can take a look and how we can utilize this to fight spammers..etc